### ISSUE TITLE
[udp] messy code is outputted on K64F

### ISSUE THREAD

**Issue by qiaojingx** [01org/zephyr.js#918]
### Description

messy code is outputted on K64F
 
### Test Code

[UDPEchoServ4.js](https://github.com/01org/zephyr.js/blob/master/samples/UDPEchoServ4.js); [UDPEchoServ6.js](https://github.com/01org/zephyr.js/blob/master/samples/UDPEchoServ6.js); [test-udp4-server-manual.js](https://github.com/01org/zephyr.js/blob/master/tests/test-udp4-server-manual.js); [test-udp6-server-manual.js](https://github.com/01org/zephyr.js/blob/master/tests/test-udp6-server-manual.js); [test-udp4-server.js](https://github.com/01org/zephyr.js/blob/master/tests/test-udp4-server.js); [test-udp6-server.js](https://github.com/01org/zephyr.js/blob/master/tests/test-udp6-server.js);
 
### Steps to Reproduction

1. `make BOARD=frdm_k64f JS=samples/UDPEchoServ4.js` 
2. cp outdir/frdm_k64f/zephyr.bin /media/<USERNAME>/MBED/

### Actual Result

![selection_016](https://cloud.githubusercontent.com/assets/19813546/24391855/ddf23bca-13c3-11e7-9ad0-a72cb7ea68e9.png)
![selection_009](https://cloud.githubusercontent.com/assets/19813546/24391860/e264d190-13c3-11e7-951f-2338beb9db91.png)
![selection_009](https://cloud.githubusercontent.com/assets/19813546/24391862/e84f443c-13c3-11e7-8aa1-bd32ca9e8970.png)
![selection_008](https://cloud.githubusercontent.com/assets/19813546/24391863/e87b502c-13c3-11e7-9507-87703fa38e35.png)
 
### Expected Result

No messy code output 

### Test Builds

Branch | Commit Id | Target Device | Test Date | Result
---|---|---|---|---
master | bb11854 | FRDM-K64F | Mar 28, 2017 | Fail

**Comment by pfalcon** [comment]
Note: I don't have access to frdm_k64f at this time (for few weeks), so won't be able to test this :-(.

**Comment by haoxli** [comment]
#386 feature verification is blocked by this issue.

**Comment by pfalcon** [comment]
So, I thought for a while what that "messy code" (garbage output) could be, and I guess I got it - that looks like SLIP being output to a (user) serial console. That reminds me of one of the decision points on critical path to further evolution of networking support in Z.js: proper compile-time (prj.conf) networking configuration. Inasmuch as I like hardware-independ config, I couldn't come up with an idea how to do such for different boards having very different network interfaces. So, in MicroPython, I reduced to having a per-BOARD config fragments to configure networking. That's the way I may suggest to Z.js too.
